operator/pkg/bgpv2/cluster.go

```diff
@@ -27,6 +29,11 @@ func (b *BGPResourceManager) reconcileBGPClusterConfigs(ctx context.Context) err
 		if rcErr != nil {
 			err = errors.Join(err, rcErr)
 		}
+
+		_, uErr := b.clientset.CiliumV2alpha1().CiliumBGPClusterConfigs().UpdateStatus(ctx, config, meta_v1.UpdateOptions{})
```

Comment:
One nit: Do we want to call API server for update if nothing has changed?

Reply:
If status has not mutated, imo we should not call UpdateStatus.

Reply:
Yes, we should definitely check that before sending.

Reply:
There's a convenience function in the `meta` library that might help: https://pkg.go.dev/k8s.io/apimachinery/pkg/api/meta#SetStatusCondition

Reply:
> If status has not mutated, imo we should not call UpdateStatus.

Yeah, definitely. I'll fix that.

> There's a convenience function in the meta library that might help:

Ah, this is nice. Thanks!